### PR TITLE
[RFR] fix cloud-init tests

### PR DESCRIPTION
**Pull request by jaryn**
## Purpose or Intent

- __Fixing__ cloud-init tests that didn't work as they needed IPv6 connectivity between the test executing machine and the VM on provider. This was solved by tunnelling to the provider host and using any IP reported by provder that is connective from the host.
__Adding__ test for DHCP configuration.

{{py.test: -sv cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py::test_provision_cloud_init_payload --long-running --use-provider complete }}

### CODE REVIEW

File: cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py [hunk 1]
```diff
@@ -154,16 +156,24 @@ def test_provision_cloud_init_payload(appliance, request, setup_provider, provid
...
-        'ip_address': '169.254.0.1',
-        'subnet_mask': '29',
-        'gateway': '169.254.0.2',
-        'dns_servers': '169.254.0.3',
-        'dns_suffixes': 'virt.lab.example.com',
         'custom_template': {'name': 'oVirt cloud-init'}
     }
 
+    THERE_IS_A_WAY_TO_CHECK_STATIC_IP_VM = False
+    # TODO on review: How? Perhaps we can check the logs from the cloud-init somehow?
```
Comment (mshriver) [inline cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py:165]: Discussed a bit in comments how we can validate this when we're setting a link-local address in cloud-init, that will make the VM unreachable from the outside.

Options included:
1. Use the host as a jump box, to reach an address in the internal RHV CIDR range selected for cloud-init's payload
2. Use multiple vNIC devices on the template/VM, which can be added with wrapanapi. Let cloud-init configure the first device, let RHV DHCP assign on the secondary device.

File: cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py [hunk 2]
```diff
@@ -154,16 +156,24 @@ def test_provision_cloud_init_payload(appliance, request, setup_provider, provid
...
-        'ip_address': '169.254.0.1',
-        'subnet_mask': '29',
-        'gateway': '169.254.0.2',
-        'dns_servers': '169.254.0.3',
-        'dns_suffixes': 'virt.lab.example.com',
         'custom_template': {'name': 'oVirt cloud-init'}
     }
 
+    THERE_IS_A_WAY_TO_CHECK_STATIC_IP_VM = True
+    # TODO remove this condition
```
Comment (mshriver) [inline cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py:165]: yes please do :)

File: cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py [hunk 3]
```diff
@@ -175,20 +185,26 @@ def test_provision_cloud_init_payload(appliance, request, setup_provider, provid
...
                                                                             partial_check=True)
     check_all_tabs(provision_request, provider)
     provision_request.wait_for_request()
 
-    connect_ip = wait_for(find_global_ipv6, func_args=[instance], num_sec=600, delay=20).out
-    logger.info(f'Connect IP: {connect_ip}')
+    wait_for(lambda: ci_payload['ip_address'] in instance.mgmt.all_ips)
+
+    vmip, = (ip for ip in instance.mgmt.all_ips if ip.startswith('2620'))
+    jump_host_config = conf.cfme_data['management_systems'][provider.key]['hosts'][0]
```
Comment (mshriver) [inline cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py:199]: You can read this directly from the `provider` instance

File: cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py [hunk 4]
```diff
@@ -175,20 +185,26 @@ def test_provision_cloud_init_payload(appliance, request, setup_provider, provid
...
+    jump_host_creds = conf.credentials[jump_host_config['credentials']['default']]
+
+    with closing(SSHClient(hostname=jump_host_config['name'],
```
Comment (mshriver) [inline cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py:202]: Why are these calls wrapped in closing? The SSHClient context manager should be handling closing the tunnel on exit. 

If its necessary for some reason, please include the description of why as a comment block here.

File: cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py (hunk 3 above)
Comment (jaryn) [reply cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py]: done

File: cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py (hunk 2 above)
Comment (jaryn) [reply cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py]: done

File: cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py (hunk 1 above)
Comment (jaryn) [reply cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py]: I used the jumpbox approach

File: cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py (hunk 4 above)
Comment (jaryn) [reply cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py]: done

File: cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py [hunk 5]
```diff
@@ -128,11 +122,125 @@ def test_provision_cloud_init(appliance, request, setup_provider, provider, prov
...
+                                      username=jump_host_creds.principal,
+                                      password=jump_host_creds.secret)
+
+        def _connection_factory(ip):
+            return jump_host_session.tunnel_to(
+                hostname=ip,
+                username='root', password=static_mode_payload['root_password'],
+                timeout=ssh.CONNECT_TIMEOUT)
+
+        # Cleanup this explicitly because we can get to problems with ordering the cleanups of
```
Comment (mshriver) [inline cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py:228]: Thanks for including this explanation of the explicit closes!

**Pull request merged**